### PR TITLE
Remove locking from Atomic App

### PR DESCRIPTION
**Pull request by cdrage**
Removed due to numerous issues with locking (dev + test environments) as well as
the deprecation of LockFile.

### CODE REVIEW

File: atomicapp/cli/main.py [hunk 1]
```diff
@@ -474,31 +471,6 @@ def run(self):
             if hasattr(args, item) and getattr(args, item) is not None:
                 args.cli_answers[item] = getattr(args, item)
 
-        lock = LockFile(os.path.join(Utils.getRoot(), LOCK_FILE))
```
Comment (cdrage) [inline atomicapp/cli/main.py]: one sec. i'll fix this

**Comment by cdrage** [comment]
#dotest

**Comment by cdrage** [comment]
#dotests

**Comment by dustymabe** [comment]
unittests fail

**Comment by cdrage** [comment]
#dotests

**Comment by cdrage** [comment]
Re-doing tests

**Comment by dustymabe** [comment]
#dotestsdocker

**Comment by dustymabe** [comment]
#dotestkubernetes

**Comment by dustymabe** [comment]
#dotestdocker

**Comment by cdrage** [comment]
All passed :+1:

**Comment by dustymabe** [comment]
> We're removing locking in favour of each provider handling the way that
> each artifact is deployed.
> 
> For example: k8s checks the namespaces, docker checks the docker name
> before deployment, oc checks either the configuration or namespace

I don't think the intent of locking was really to ever do anything other than to prevent multiple instances of atomicapp from running at the same time (similar to the way yum prevents multiple instances of itself from running at the same time). So your statement doesn't really apply.

If anything we are simply removing locking now because it is causing more problems than it solves. If we get to a point where people are scripting the use of atomicapp and have issues with multiple instances running at the same time then we will add it back.

**Comment by cdrage** [comment]
@dustymabe 

Updated the PR as well as commit message.

**Comment by dustymabe** [comment]
#dotests

**Comment by dustymabe** [comment]
#dotestkubernetes

**Comment by cdrage** [comment]
Looks to be timeout issues / timing issues again. Going to run this locally as well and see what I get.

**Comment by dustymabe** [comment]
@cdrage LGTM.. if you can get the tests to pass then merge away.

**Comment by cdrage** [comment]
#dotests

**Comment by dustymabe** [comment]
#dotestkubernetes

**Pull request merged**